content/blog/2022/2022-01-25-annotation-processing.md

```diff
@@ -0,0 +1,951 @@
+
+Normally when we run JavaDoc on the class CustomAnnotatedManager the annotation information would not show up in the documentation. But when we use the @Documented annotation as below:
+
+```
+@Inherited
+@Documented
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface Company{
+    String name() default "ABC";
+    String city() default "XYZ";
+}
+```
+
+ ...the annotation information would show up in the output too, as below:
+
+```java
+Package com.reflectoring
+Class CustomAnnotatedEmployee
+
+java.lang.Object
+com.reflectoring.CustomAnnotatedEmployee
+
+@Company
+public class CustomAnnotatedEmployee
+extends java.lang.Object
+
+Constructor Summary
+Constructors
+
+Constructor	Description
+CustomAnnotatedEmployee​(int id, java.lang.String name)
+```
+
+#### @Repeatable
+
+@Repeatable allows multiple repeating Custom Annotations on a method, class, or field. To use a Repeatable Annotation, we need to wrap the Annotation in a Container class which refers to it as an array, as below:
+
+```java
+@Target(ElementType.TYPE)
+@Repeatable(RepeatableCompanies.class)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface RepeatableCompany {
+    String name() default "Name_1";
+    String city() default "City_1";
+}
+```
+
+```java
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface RepeatableCompanies {
+    RepeatableCompany[] value() default{};
+}
+```
+
+We declare our main Class as below:
+
+```java
+@RepeatableCompany
+@RepeatableCompany(name =  "Name_2", city = "City_2")
+public class RepeatedAnnotatedEmployee {
+}
+```
+
+If we run a test on it as below:
+
+```java
+public class TestRepeatedAnnotation {
+
+    public static void main(String[] args) {
+
+        RepeatableCompany[] repeatableCompanies = RepeatedAnnotatedEmployee.class.getAnnotationsByType(RepeatableCompany.class);
+        for (RepeatableCompany repeatableCompany : repeatableCompanies) {
+
+            System.out.println("Name: " + repeatableCompany.name());
+            System.out.println("City: " + repeatableCompany.city());
+        }
+    }
+}
+```
+
+We get the following output which displays the value of multiple annotations. Both the default annotation values and the custom values we provided are printed out.
+
+```
+Name: Name_1
+City: City_1
+Name: Name_2
+City: City_2
+```
+
+#### @Target
+
+@Target specifies at which element the Annotation can be used, for eg in the above example the annotation Company was defined only for TYPE and so it could only be applied to a Class.
+
+```java
+@Company
+public class Employee {
+
+    @Company
+    public void getEmployeeStatus(){
+
+        System.out.println("This is the Base Employee class");
+    }
+}
+```
+
+If we apply @Company to the method 'getEmployeeStatus' as above, we get a compiler error stating: "'@Company' not applicable to method."
+
+The various self-explanatory Target types are:
+
+​		ElementType.ANNOTATION_TYPE, 
```

Comment:
Please use a Markdown-formatted list, i.e. like this:

* a
* b
* c

You can see a preview of the article here: https://deploy-preview-446--reflectoring.netlify.app/blog/2022/2022-01-25-annotation-processing/. This preview is automatically updated after you push changes to your branch, so you can check your changes.